Cognite.Simulator.Tests/UtilsTests/LicenseControllerTest.cs

```diff
@@ -0,0 +1,176 @@
+            
+            // Act
+            tracker.AcquireLicense();
+            Assert.Equal(TestLicenseState.Held, license.State);
+            Assert.True(tracker.LicenseHeld);
+            using (var usage = tracker.BeginUsage())
+            {
+                Thread.Sleep(200); // Wait longer than lock time
+                Assert.True(tracker.LicenseHeld); // Should not release while in use
+            }
+
+            tracker.Dispose();
+        }
+
+        [Fact]
+        public void LicenseTracker_ShouldResetTimer_OnNewUsage()
+        {
+            // Arrange
+            var licenseLockTime = TimeSpan.FromMilliseconds(100);
+            var (tracker, license) = CreateTracker(licenseLockTime: licenseLockTime);
+
+            // Act  
+            tracker.AcquireLicense();
+            Assert.Equal(TestLicenseState.Held, license.State);
+            using (tracker.BeginUsage()) { } // First usage
+            Thread.Sleep(50);
+            
+            using (tracker.BeginUsage()) { } // Second usage should reset timer
+            Thread.Sleep(50);
+            
+            // Assert
+            Assert.True(tracker.LicenseHeld); // Should not be released yet due to timer reset
+
+            Thread.Sleep(150); // Wait for full lock time + buffer
+            Assert.True(license.State == TestLicenseState.Released);
+            Assert.False(tracker.LicenseHeld);
+
+            tracker.Dispose();
+        }
+
+        [Fact]
+        public void LicenseTracker_ShouldPreventRelease_DuringContinuousUsage()
+        {
+            // Arrange
+            var (tracker, license) = CreateTracker();
+
+            // Act
+            tracker.AcquireLicense();
+            using (var usage1 = tracker.BeginUsage())
+            {
+                Thread.Sleep(50);
+                using (var usage2 = tracker.BeginUsage())
+                {
+                    Thread.Sleep(100);
+                    // Should not release during overlapping usage
+                    Assert.True(tracker.LicenseHeld);
+                }
+            }
+
+            Thread.Sleep(200); // Wait longer than lock time after all usage ends
+            
+            // Assert
+            Assert.True(license.State == TestLicenseState.Released);
+            tracker.Dispose();
+        }
+
+        [Fact]
+        public void LicenseTracker_ShouldUpdateState_WhenLicenseReleasedManually()
+        {
+            // Arrange
+            var (tracker, license) = CreateTracker();
+            tracker.AcquireLicense();
+            using (tracker.BeginUsage()) { }
+            Assert.True(tracker.LicenseHeld);
+
+            // Act
+            tracker.ReleaseLicenseWithoutCallback();
+            
+            // Assert
+            Assert.False(tracker.LicenseHeld);
+            // Should not change external license state (since this is a manual release)
+            Assert.True(license.State == TestLicenseState.Held);
```

Comment:
check if there's timer precision thing we could use

Reply:
seems like not, unless different kind of timers are used